hbase-server/src/main/java/org/apache/hadoop/hbase/mob/MobUtils.java

```diff
@@ -314,20 +316,24 @@ public static void cleanExpiredMobFiles(FileSystem fs, Configuration conf, Table
           }
           filesToClean
             .add(new HStoreFile(fs, file.getPath(), conf, cacheConfig, BloomType.NONE, true));
+          if (
+            filesToClean.size() > conf.getInt(MOB_CLEANER_BATCH_SIZE_UPPER_BOUND,
+              DEFAULT_MOB_CLEANER_BATCH_SIZE_UPPER_BOUND)
+          ) {
+            try {
+              removeMobFiles(conf, fs, tableName, mobTableDir, columnDescriptor.getName(),
+                filesToClean);
+              LOG.info("Table {} {} expired mob files are deleted", tableName, filesToClean.size());
+            } catch (IOException e) {
+              LOG.error("Failed to delete the mob files, table {}", tableName, e);
+            }
+            filesToClean.clear();
+          }
         }
       } catch (Exception e) {
         LOG.error("Cannot parse the fileName " + fileName, e);
       }
     }
-    if (!filesToClean.isEmpty()) {
```

Comment:
We still need this piece of code?  For the last round, where we do not have 10000 files, we will exit the loop with an unempty filesToClean?

Reply:
Got it, thanks.